Refactoring to TypeScript and Project Configuration Updates

ad-astra-per-ardua opened the pull request:
![image](https://github.com/Webwiznitr/Hacktoberfest-Tulip/assets/50827253/a8e85e76-3707-41c3-b47e-177f0af7b569)

**Summary of Changes:**

1. index.js and routes.js have been transformed to index.ts and routes.ts, respectively.
2. TypeScript-related settings have been added to tsconfig.json.
3. Several packages have been added to package.json.

**Detailed Changes:**

TypeScript Refactoring:

- The files index.js and routes.js have been changed to .ts extensions.
- Explicit types have been specified for the Express application and router variables.

TypeScript Configuration:

- The tsconfig.json file has been added to configure the TypeScript project.
- Key configurations include the compilation target version, libraries used, and the module system.


Dependency Updates:

- Several frontend-related version packages and the typescript package have been added.
- TypeScript-related scripts have been added to the execution scripts.


I kindly ask for a thorough review. Please share any thoughts or suggestions you might have. :>

ad-astra-per-ardua committed:
Add ts setting files.

ad-astra-per-ardua committed:
The versions in the server's package.json have been harmonized for consistency with client ver.

ad-astra-per-ardua committed:
Convert JS files to TypeScript

ad-astra-per-ardua committed:
Modified with packages.

ad-astra-per-ardua commented:
#14

AmreetKumarkhuntia commented:
Will check it out

AmreetKumarkhuntia commented on server/src/index.ts:
```diff
@@ -21,13 +21,14 @@ app.use(express.json());
 // Start the Express server
 app.listen(port, () => {
   console.log(`Server is running on port ${port}`);
+  console.log(`Refactoring Completed.`);
```
Remove this console

AmreetKumarkhuntia commented on server/package.json:
```diff
@@ -1,23 +1,62 @@
     "helmet": "^7.0.0",
     "morgan": "^1.10.0",
-    "nodemon": "^3.0.1"
+    "react": "^18.2.0",
```
Remove react dependencies.

AmreetKumarkhuntia commented on server/package.json:
```diff
@@ -1,23 +1,62 @@
 {
```
looks like this is copied from client

AmreetKumarkhuntia reviewed (commented):
Do not replace the package.json from client package

AmreetKumarkhuntia reviewed (changes requested):
Work on the comments and do not replace the package.json file.

ad-astra-per-ardua commented:
> Work on the comments and do not replace the package.json file.

Is it okay to change the start script from "start": "node src/index.ts" to "start": "ts-node src/index.ts" in our development setup?

AmreetKumarkhuntia commented:
> > Work on the comments and do not replace the package.json file.
> 
> Is it okay to change the start script from "start": "node src/index.ts" to "start": "ts-node src/index.ts" in our development setup?

I don't think it's necessary but ya can keep it. Also, add the required dev dependencies for that to work.

ad-astra-per-ardua committed:
Convert JS files to TypeScript

ad-astra-per-ardua committed:
Modified tsconfig.json to ensure optimal TypeScript configuration for our server-side environment.

ad-astra-per-ardua committed:
Ensured that only server-related dependencies are present, avoiding any mix-up with the client-side package.json

ad-astra-per-ardua committed:
resolve conflict

ad-astra-per-ardua commented:
![image](https://github.com/Webwiznitr/Hacktoberfest-Tulip/assets/50827253/48460af8-6a22-4d7a-988a-a72e4a6be5db)
![image](https://github.com/Webwiznitr/Hacktoberfest-Tulip/assets/50827253/f44a5fa1-e418-4715-9af8-9dd4f4d1bf11)

- Adjusted the tsconfig.json to better suit the server environment and the use of TypeScript.
- Updated the package.json to reflect the necessary dependencies and scripts for TypeScript. Removed any unrelated dependencies, ensuring it's distinct from the client's package.json.

ad-astra-per-ardua commented:
It was honored :)
